pkg/api/internalclient/firewall.go

```diff
@@ -0,0 +1,47 @@
+package internalclient
+
+import (
+	"errors"
+	"net"
+	"net/http"
+
+	"github.com/go-resty/resty/v2"
+)
+
+// firewallAPI defines methods for interacting with firewall-related functionality.
+type firewallAPI interface {
+	// FirewallEvaluate evaluates firewall rules based on the provided lookup parameters.
+	// It returns an error if the evaluation fails or if a firewall rule prohibits the connection.
+	FirewallEvaluate(lookup map[string]string) error
+}
+
+var (
+	ErrFirewallConnection = errors.New("failed to make the request to evaluate the firewall")
+	ErrFirewallBlock      = errors.New("a firewall rule prohibit this connection")
+)
+
+func (c *client) FirewallEvaluate(lookup map[string]string) error {
+	local := resty.New()
+	local.AddRetryCondition(func(r *resty.Response, err error) bool {
+		if _, ok := err.(net.Error); ok {
+			return true
+		}
+
+		return r.StatusCode() >= http.StatusInternalServerError && r.StatusCode() != http.StatusNotImplemented
+	})
```

Comment:
I believe we should do something like this on billing requests, no?

Reply:
I haven't made any changes to the implementation within the existing methods. The way `BillingReport` and `BillingEvaluate` function remains the same as before these commits.

Reply:
You are right, but the usage of base URL could cause a small issue in URL on `billing` routes. Firewall ones are immune to it because it creates it own client.

Reply:
Since the base URL won't cause this issue, I don't believe creating a new client here is a good approach. It would result in duplicating some code snippets in four places, which could lead to minimal bugs that are difficult to identify if changes are made in one place but not in another. Additionally, I also believe that the firewall should utilize the client's HTTP instead of creating a new one.